Sources/macro.js

```diff
@@ -360,7 +360,11 @@ export function setArray(publicAPI, model, fieldNames, size) {
         array = array[0];
       }
 
-      if (array.length !== size) {
+      if (Array.isArray(size)) {
+        if (!size.includes(array.length)) {
+          throw new RangeError('Invalid number of values for array setter');
+        }
+      } else if (array.length !== size) {
```

Comment:
Sorry, I hadn't got here yet.  I guess your stuff above could work, though I might have preferred a new renderer property "backgroundOpacity" instead of trying to support setters for arrays of variable length.  I wonder what @jourdain thinks?